### PR TITLE
Add test cases for hierarchical package names

### PR DESCRIPTION
**Pull request by Dilhasha**
## Purpose
Add test cases for hierarchical package names.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java [hunk 1]
```diff
@@ -199,4 +229,148 @@ static Path getBalaPath(Path projectPath, String org, String pkgName, String pla
...
+     * Search for the given package on Central.
+     *
+     * @param pkg package
+     * @param tempWorkspaceDirectory Path to workspace
+     * @param envVariables Environmental variables
+     * @return
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    private static String searchPackageDetails(String pkg, Path tempWorkspaceDirectory,
```
Comment (azinneera) [inline project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java:277]: Won't this be useful for other tests as well? If so, we can make it package-private/public

File: project-api-tests/src/test/java/org/ballerina/projectapi/HierarchicalPackagesTest.java [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+import static org.ballerina.projectapi.CentralTestUtils.getEnvVariables;
+import static org.ballerina.projectapi.CentralTestUtils.getExecutableJarPath;
+import static org.ballerina.projectapi.CentralTestUtils.getString;
+import static org.ballerina.projectapi.CentralTestUtils.testPushPackage;
+import static org.ballerina.projectapi.CentralTestUtils.testPushPackageToLocal;
+import static org.ballerina.projectapi.TestUtils.DISTRIBUTION_FILE_NAME;
+import static org.ballerina.projectapi.TestUtils.OUTPUT_CONTAIN_ERRORS;
+import static org.ballerina.projectapi.TestUtils.executeBuildCommand;
+
+public class HierarchicalPackagesTest {
```
Comment (azinneera) [inline project-api-tests/src/test/java/org/ballerina/projectapi/HierarchicalPackagesTest.java:41]: Missing class comment

File: project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java (hunk 1 above)
Comment (Dilhasha) [reply project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java:277]: This method contains reused logic used by other package-private/public methods within the same class. So, IMO no need to make this available.

File: project-api-tests/src/test/java/org/ballerina/projectapi/BuildTimeTest.java [hunk 3]
```diff
@@ -26,6 +26,7 @@
 import static org.ballerina.projectapi.CentralTestUtils.getEnvVariables;
 import static org.ballerina.projectapi.CentralTestUtils.getString;
 import static org.ballerina.projectapi.TestUtils.DISTRIBUTIONS_DIR;
```
Comment (pramodya1994) [inline project-api-tests/src/test/java/org/ballerina/projectapi/BuildTimeTest.java:28]: There are some unused imports here

File: project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java [hunk 4]
```diff
@@ -27,13 +27,25 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Arrays;
```
Comment (pramodya1994) [inline project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java:31]: Unused imports

File: project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java [hunk 5]
```diff
@@ -163,10 +181,22 @@ static String getGenerateBalaLog(String org, String pkgName, String platform, St
         return "Creating bala\n" + "\ttarget/bala/" + org + "-" + pkgName + "-" + platform + "-" + version + ".bala";
     }
 
+    /**
+     * Get the log output when package is successfully pushed to Central.
+     *
+     * @param orgName Organization name
+     * @param pkgName
```
Comment (pramodya1994) [inline project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java:188]: Shall we add a description for `pkgName`?
Is it okay to keep `return` empty in L190?

File: project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java [hunk 6]
```diff
@@ -199,4 +229,148 @@ static Path getBalaPath(Path projectPath, String org, String pkgName, String pla
...
+                                           Map<String, String> envVariables) throws IOException, InterruptedException {
+        Process search = executeSearchCommand(DISTRIBUTION_FILE_NAME,
+                tempWorkspaceDirectory,
+                new LinkedList<>(Collections.singletonList(pkg)),
+                envVariables);
+        String buildErrors = getString(search.getErrorStream());
+        if (!buildErrors.isEmpty()) {
+            Assert.fail(OUTPUT_CONTAIN_ERRORS + buildErrors);
+        }
+        String buildOutput = getString(search.getInputStream());
```
Comment (pramodya1994) [inline project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java:287]:
```suggestion
        return getString(search.getInputStream());
```
We can directly return right?

File: project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java [hunk 7]
```diff
@@ -199,4 +229,148 @@ static Path getBalaPath(Path projectPath, String org, String pkgName, String pla
...
+     * @param envVariables Environmental variables
+     * @param packageName Package
+     * @param orgName Organization name
+     * @param version Package Version
+     * @param additionalArgs Additional arguments to pass wben building package.
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    static void buildPackageBala(Path tempWorkspaceDirectory, Map<String, String> envVariables, String packageName,
+                             String orgName, String version, List<String> additionalArgs) throws IOException, InterruptedException {
```
Comment (pramodya1994) [inline project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java:304]: Line is too long

File: project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java [hunk 8]
```diff
@@ -199,4 +229,148 @@ static Path getBalaPath(Path projectPath, String org, String pkgName, String pla
...
+     * @param projectName Project name
+     * @param envVariables Environmental variables
+     * @param orgName Organization name
+     * @param packageName package
+     * @param version Package Version
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    static void testPushPackage(Path tempWorkspaceDirectory, String projectName, Map<String, String> envVariables,
+                                String orgName, String packageName, String version) throws IOException, InterruptedException {
```
Comment (pramodya1994) [inline project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java:338]: Line is too long

Comment (pramodya1994) [reply project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java:338]: There are few places to correct

Comment (pramodya1994) [reply project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java:338]: Why are these styling issues not getting validated? It's better to add early as possible right?

Comment (Dilhasha) [reply project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java:338]: Looks like we haven't included check style plugin in the project. Will be incorporating it. That's the reason for missing these styling issues.

File: project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java (hunk 7 above)
Comment (Dilhasha) [reply project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java]: Fixed with ed75e28e7d3262e29f98398857c356cf789391af

File: project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java (hunk 6 above)
Comment (Dilhasha) [reply project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java]: Fixed with 7189e44c00191cfb0e74d468ffcc2b266bdc40c7

File: project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java (hunk 4 above)
Comment (Dilhasha) [reply project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java]: Fixed with e697e99dbd864835f445fa6942c2f2b0792c64b5

File: project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java (hunk 5 above)
Comment (Dilhasha) [reply project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java]: Updated with e697e99dbd864835f445fa6942c2f2b0792c64b5

File: project-api-tests/src/test/java/org/ballerina/projectapi/BuildTimeTest.java (hunk 3 above)
Comment (Dilhasha) [reply project-api-tests/src/test/java/org/ballerina/projectapi/BuildTimeTest.java]: Fixed with e697e99dbd864835f445fa6942c2f2b0792c64b5

**Review by azinneera — APPROVED** [review]

**Pull request merged**